vispy/visuals/mesh.py

```diff
@@ -21,16 +22,22 @@
 
 vertex_template = """
 varying vec4 v_base_color;
+varying float v_is_shown;
 
 void main() {
+    v_is_shown = $clip_by_planes($position);
     v_base_color = $color_transform($base_color);
     gl_Position = $transform($to_vec4($position));
 }
 """
 
 fragment_template = """
 varying vec4 v_base_color;
+varying float v_is_shown;
+
 void main() {
+    if (v_is_shown < 0.)
+        discard;
     gl_FragColor = v_base_color;
 }
```

Comment:
This looks to me like a `pre` hook for the vertex shader, and a `pre` hook for the fragment shader. Although both of them could be `post` hooks too. These would both be added in one Filter.

There are some filters in the color.py filters that I pointed to that use `position` so you could probably borrow from that logic to get `$position` usable in your filter code.

Reply:
Oh, sweet!